### PR TITLE
[BACKEND] B-get-product-list

### LINKED ISSUE #403: [BACKEND] Implement Get Product Lists endpoint
All product lists should be returned and product serializer is used for this purpose.

### PR DESCRIPTION
**Pull request by bekir96**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Description

This PR includes get product lists endpoint. Response body is:

image
image

id represents pk of  product list

# Test results

image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ecommerce/backend/API/views/product_list.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        product_list = ProductList(user_id=user.pk, name=list_name)
+        product_list.save()
+
+        return Response({'id': product_list.pk, 'status': { 'successful': True, 'message': "List is successfully created."}})
+
+    elif request.method == "GET":
+        response_product_list = []
+        product_lists = ProductList.objects.filter(user_id=user.pk)
+
```
Comment (omerfdeniz) [inline ecommerce/backend/API/views/product_list.py:34]: Check the edge case if no such product_list exists.

```suggestion
if not product_lists:
    return Response({'status': {'successful': False, 'message':'No such list exists.'})

```

**Review by omerfdeniz — CHANGES_REQUESTED** [review]
Left a comment to address before merge, other than that good work again

File: ecommerce/backend/API/views/product_list.py (hunk 1 above)
Comment (umutoksuz) [reply ecommerce/backend/API/views/product_list.py:34]: This is not a query function, this retrieves all lists of a user by user's id. So, if user has no lists created, it simply returns an empty list. Edge case may not be a problem.

**Review by umutoksuz — APPROVED** [review]
Looks good to me!

File: ecommerce/backend/API/views/product_list.py (hunk 1 above)
Comment (furkannane) [reply ecommerce/backend/API/views/product_list.py:34]: I think Umut is right. It is not a problem. Frontend may check if the list is empty and then they can print something like "you have no list".

**Review by furkannane — APPROVED** [review]
Nicely done, I saw no problem.

File: ecommerce/backend/API/views/product_list.py (hunk 1 above)
Comment (bekir96) [reply ecommerce/backend/API/views/product_list.py:34]: Actually, I traverse all product_lists with for loop, so if no item in it, response will return empty

**Review by bekir96 — COMMENTED** [review]
the suggestion has been checked and there is no need for changes.

**Review by omerfdeniz — APPROVED** [review]

**Pull request merged**